src/main/java/org/example/learningsystem/core/destination/service/DestinationServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package org.example.learningsystem.core.destination.service;
+
+import lombok.RequiredArgsConstructor;
+import org.example.learningsystem.core.destination.config.DestinationServiceProperties;
+import org.example.learningsystem.core.destination.dto.AccessTokenResponseDto;
+import org.example.learningsystem.core.destination.dto.SmtpDestinationResponseDto;
+import org.example.learningsystem.email.config.EmailServerProperties;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestClient;
+
+import java.util.Map;
+
+import static java.util.Objects.nonNull;
+
+@Service
+@RequiredArgsConstructor
+@Profile("cloud")
+public class DestinationServiceImpl implements DestinationService {
+
+    private final DestinationServiceProperties destinationServiceProperties;
+    private final RestClient restClient = RestClient.create();
```

Comment:
`RestTemplate` should not be created for every service idependently, it should be configured as singleton bean and reused for all requests in all services (in your case)